### ISSUE TITLE
ebiten: `MaximizeWindow` does not work even when maximum window size is not limited

### ISSUE THREAD

**Issue by erexo** [hajimehoshi/ebiten#2290]
In 2.4.0 #2259 tried to fix unexpected window flickering when window size limits are determined, this does not work properly because maximization should not depend on the window's min size. If a game determined minimum without restriction of maximum size, `ebiten.MaximizeWindow()` should still be available.

```
package main

import "github.com/hajimehoshi/ebiten/v2"

func main() {
	ebiten.SetWindowResizingMode(ebiten.WindowResizingModeEnabled)
	ebiten.SetWindowSizeLimits(50, 50, -1, -1)
	ebiten.MaximizeWindow() // does not work
	ebiten.RunGame(&game{})
}

type game struct{}

func (game) Update() error {
	return nil
}

func (game) Draw(screen *ebiten.Image) {
}

func (game) Layout(outsideWidth, outsideHeight int) (int, int) {
	return outsideWidth, outsideHeight
}
```

**Comment by hajimehoshi** [comment]
Thanks, I'll take a look later!

**Comment by erexo** [comment]
it is resolved in #2291 but I forgot to mention it in the commit
